sdk/src/wallet/operations/reissue.rs

```diff
@@ -66,76 +87,43 @@ where
+                    TransactionState::Accepted | TransactionState::Confirmed | TransactionState::Finalized => {
+                        return Ok(block_id);
                     }
-                    // Only reissue latest attachment of the block
-                    if index == block_ids_len - 1 && block_metadata.block_state == BlockState::Rejected {
-                        let reissued_block = self
-                            .client()
-                            .build_basic_block(
-                                todo!("issuer id"),
-                                Some(Payload::SignedTransaction(Box::new(transaction.payload.clone()))),
-                            )
-                            .await?
-                            .sign_ed25519(
-                                &*self.get_secret_manager().read().await,
-                                self.bip_path().await.ok_or(Error::MissingBipPath)?,
-                            )
-                            .await?;
-                        block_ids.push(reissued_block.id(&protocol_parameters));
+                    TransactionState::Failed => failed = true,
+                    TransactionState::Pending => {}
+                };
+            }
+            // Check if the transaction got reissued in another block and confirmed there
+            if failed {
+                let included_block = self.client().get_included_block(transaction_id).await.map_err(|e| {
+                    if matches!(e, ClientError::Node(crate::client::node_api::error::Error::NotFound(_))) {
+                        // If no block was found with this transaction id, then it couldn't get accepted
+                        ClientError::TransactionAcceptance(transaction_id.to_string())
+                    } else {
+                        e
                     }
-                }
-                // After we checked all our reissued blocks, check if the transaction got reissued in another block
-                // and confirmed
-                // TODO: can this still be the case? Is the TransactionState per transaction or per attachment in a
-                // block?
-                if failed {
-                    let included_block = self.client().get_included_block(transaction_id).await.map_err(|e| {
-                        if matches!(e, ClientError::Node(crate::client::node_api::error::Error::NotFound(_))) {
-                            // If no block was found with this transaction id, then it can't get included
-                            ClientError::TangleInclusion(first_block_id.to_string())
-                        } else {
-                            e
-                        }
-                    })?;
-                    return Ok(included_block.id(&protocol_parameters));
-                }
+                })?;
+                let protocol_parameters = self.client().get_protocol_parameters().await?;
+                return Ok(included_block.id(&protocol_parameters));
```

Comment:
maybe we can put this in an anonymous function, for example:
```rust
let handle_failed = || { ... }
```
and call this in L109 instead of this `failed` boolean.

Reply:
I'm not sure what that's supposed to achieve, can you maybe do a full code suggestion?

Reply:
hmm, had a 2nd look and what I suggested is not even necessary. Can you not just move that code executed if `failed == true` into the `TransactionState::Failed` match branch itself?

Reply:
Makes sense, done